overleaf/overleaf
# [misc] decaff and config cleanup

## Description


For https://github.com/overleaf/issues/issues/3225
For https://github.com/overleaf/issues/issues/3238
Chained onto https://github.com/overleaf/overleaf/pull/910 (remove migrations)

This PR is a big one, decaffeinate all the things, upgrade the settings module, lots of reference cleanup to get overrides of the `SHARELATEX_CONFIG` going in Server Pro and fixes for bin scripts that were parsing coffee-script with sed+grep :facepalm: . Latest web/master has dropped support for coffee-script, so opening multiple PRs for these changes do not help in verifying them as manual testing is not possible unless all the changes are applied. Sorry for that.

## Related issues / Pull Requests


For https://github.com/overleaf/issues/issues/3225
For https://github.com/overleaf/issues/issues/3238
Chained onto https://github.com/overleaf/overleaf/pull/910 (remove migrations)

## Manual testing performed

- start server ce, login, compile
- run server pro acceptance tests (saml is broken for unrelated changes, see slack https://digital-science.slack.com/archives/C20TZCMMF/p1625742371291800)
